### PR TITLE
Add commands for updating `studio.url` in DVC configuration

### PR DESCRIPTION
**Pull request by julieg18**
## Demo

~https://github.com/iterative/vscode-dvc/assets/43496356/3327633a-e747-42c9-84d2-2909175afafe~

https://github.com/iterative/vscode-dvc/assets/43496356/c2886dea-8f51-4931-b273-ed7455c5f68d

Part of #5035

### CODE REVIEW

File: extension/package.json [hunk 1]
```diff
@@ -396,6 +401,11 @@
         "command": "dvc.selectFocusedProjects",
         "category": "DVC"
       },
+      {
+        "title": "Set Self-Hosted Studio URL",
```
Comment (julieg18) [inline extension/package.json:405]: Decided on just using one command instead of `Update` and `Add` commands.

File: extension/src/setup/index.ts [hunk 2]
```diff
@@ -325,6 +325,30 @@ export class Setup
...
+  public removeStudioUrl() {
+    if (!this.getCliCompatible()) {
+      return
+    }
+
+    return this.studio.removeStudioUrl(this.dvcRoots)
+  }
+
+  public async saveStudioUrl() {
+    const cwd = this.getCwd()
```
Comment (julieg18) [inline extension/src/setup/index.ts:337]: Did look into using `getCwd()` for `saveStudioAccessToken` and it does look like we can use it without issues. Will update (plus maybe remove the dvc compatible check as well) in a followup!

File: extension/package.json (hunk 1 above)
Comment (julieg18) [reply extension/package.json:405]: Ohhhh, ok. Well, if we want all commands to be consistent, `Add` makes more sense to me in other commands contexts. I'll go ahead and just use `Update` and `Add` commands 😄 for this too.

File: extension/src/test/suite/setup/index.test.ts [hunk 3]
```diff
@@ -1213,6 +1211,121 @@ suite('Setup Test Suite', () => {
       )
     }).timeout(WEBVIEW_TEST_TIMEOUT)
 
+    it("should be able to save the user's self-hosted url", async () => {
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply extension/src/test/suite/setup/index.test.ts:1214]: [C] When you are making multiple chained assertions it is good to add a description to the `expect` statements. E.g.

```
      expect(
        executeCommandSpy,
        'should call executeCommand with the args required to select a file to compare against'
      ).to.be.calledWith('selectForCompare', baseline)
```

`expect(saveStudioUrlSpy).not.to.be.called` would be a good place to add one as this has nothing to do with the `it` statement 

and/or split the test

**Review by mattseddon — APPROVED** [review]

File: extension/src/test/suite/setup/index.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (julieg18) [reply extension/src/test/suite/setup/index.test.ts:1214]: > [C] When you are making multiple chained assertions it is good to add a description to the expect statements.

Good idea! I forget that we can add descriptions to expect statements, thanks!

File: extension/src/setup/index.ts (hunk 2 above)
Comment (julieg18) [reply extension/src/setup/index.ts:337]: Discussed synchronously, but we'll keep the dvc compatible check for failsafe reasons.

**Pull request merged**